PWGCF/EbyEFluctuations/Tasks/nfMomentsTask.cxx

```diff
@@ -0,0 +1,389 @@
+
+  void init(o2::framework::InitContext&)
+  {
+    AxisSpec axisPt[5] = {{100, -0.01, 3 * confPtBins.value[1], ""}, {100, -0.01, 3 * confPtBins.value[3], ""}, {100, -0.01, 3 * confPtBins.value[5], ""}, {100, -0.01, 3 * confPtBins.value[7], ""}, {100, -0.01, 3 * confPtBins.value[9], ""}}; // pT axis
+
+    if (confNumPt != static_cast<int>(confPtBins.value.size()) / 2) {
+      throw std::runtime_error("Number of pT bins in confNumPt does not match the number of pT cuts in config.json!");
+    }
+    for (auto iM = 0; iM < 40; ++iM) {
+      binningM[iM] = 2 * (iM + 2);
+    }
+    for (auto iPt = 0; iPt < confNumPt; ++iPt) {
+      mHistArrQA.push_back(std::get<std::shared_ptr<TH1>>(histos.add(Form("bin%i/mEta", iPt + 1), Form("#eta for bin %.2f-%.2f;#eta", confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH1F, {{1000, -2, 2}})));
+      mHistArrQA.push_back(std::get<std::shared_ptr<TH1>>(histos.add(Form("bin%i/mPt", iPt + 1), Form("pT for bin %.2f-%.2f;pT", confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH1F, {axisPt[iPt]})));
+      mHistArrQA.push_back(std::get<std::shared_ptr<TH1>>(histos.add(Form("bin%i/mPhi", iPt + 1), Form("#phi for bin %.2f-%.2f;#phi", confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH1F, {{1000, 0, 2 * TMath::Pi()}})));
+      mHistArrQA.push_back(std::get<std::shared_ptr<TH1>>(histos.add(Form("bin%i/mMultiplicity", iPt + 1), Form("Multiplicity for bin %.2f-%.2f;Multiplicity", confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH1F, {{1000, 0, 8000}})));
+      for (auto iM = 0; iM < 40; ++iM) {
+        auto mHistsR = std::get<std::shared_ptr<TH2>>(histos.add(Form("bin%i/Reset/mEtaPhi%i", iPt + 1, iM), Form("#eta#phi_%i for bin %.2f-%.2f;#eta;#phi", iM, confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH2F, {{binningM[iM], -0.8, 0.8}, {binningM[iM], 0, 2 * TMath::Pi()}}));
+        mHistArrReset.push_back(mHistsR);
+        auto mHistsUR = std::get<std::shared_ptr<TH2>>(histos.add(Form("bin%i/UReset/mEtaPhi%i", iPt + 1, iM), Form("#eta#phi_%i for bin %.2f-%.2f;#eta;#phi", iM, confPtBins.value[2 * iPt], confPtBins.value[2 * iPt + 1]), HistType::kTH2F, {{binningM[iM], -0.8, 0.8}, {binningM[iM], 0, 2 * TMath::Pi()}}));
+        mHistArrUReset.push_back(mHistsUR);
+      }
+    }
+  }
+
+  template <class T>
+  void checkpT(const T& track)
+  {
+    for (auto iPt = 0; iPt < confNumPt; ++iPt) {
+      if (track.pt() > confPtBins.value[2 * iPt] && track.pt() < confPtBins.value[2 * iPt + 1]) {
+        mHistArrQA[iPt * 4]->Fill(track.eta());
+        mHistArrQA[iPt * 4 + 1]->Fill(track.pt());
+        mHistArrQA[iPt * 4 + 2]->Fill(track.phi());
+        countTracks[iPt]++;
+        for (auto iM = 0; iM < 40; ++iM) {
+          mHistArrReset[iPt * 40 + iM]->Fill(track.eta(), track.phi());
+          mHistArrUReset[iPt * 40 + iM]->Fill(track.eta(), track.phi());
```

Comment:
There are 2*40 histograms storing exactly the same information
Is that what is intended?

Reply:
The ones that are reset per event `mHistArrReset` are needed, but i can make `mHistArrUReset`  optional.

Reply:
But then, storing the same information on 40 histograms is what is intended?

Reply:
yes, it is :)